File: 83. Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        if not head or not head.next:
            return head
        dh = ListNode(head.val-1)
        dh.next = head
        while head:
            while head.next and head.val == head.next.val:
                head.next = head.next.next
            head = head.next
        return dh.next

File: 83. Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode prev=null, node=head;
        while (node != null) {
            prev = node;
            node = node.next;
            while (node != null && node.val == prev.val){
                node = node.next;
            }
            prev.next = node;
        }
        return head;
    }
}